# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1365922083 -7200
#      Sun Apr 14 08:48:03 2013 +0200
# Node ID 87b78e75a2b9af9c4963c2954d922a4ad5bdaef4
# Parent  db045633405c79960e5728ae4c6fd0c0213010c9
gui: undo disabling menu entries in context menu of file browser

* files-dock-widget.cc(contextmenu_requested): icon for setting file browser
  directory to octave dir, undo disabling menu entries that are not applicable
  instead of hiding them

diff --git a/libgui/src/files-dock-widget.cc b/libgui/src/files-dock-widget.cc
--- a/libgui/src/files-dock-widget.cc
+++ b/libgui/src/files-dock-widget.cc
@@ -280,47 +280,47 @@ files_dock_widget::contextmenu_requested
 
   QModelIndex index = _file_tree_view->indexAt (mpos);
   //QAbstractItemModel *m = _file_tree_view->model ();
 
   if (index.isValid())
     { 
       QFileInfo info = _file_system_model->fileInfo(index);
 
-      menu.addAction(QIcon(":/actions/icons/fileopen.png"), tr("Open"),
+      menu.addAction (QIcon (":/actions/icons/fileopen.png"), tr("Open"),
                      this, SLOT(contextmenu_open(bool)));
-      QAction *run_action = menu.addAction(
-                     QIcon(":/actions/icons/artsbuilderexecute.png"), tr("Run"),
-                     this, SLOT(contextmenu_run(bool)));
-      run_action->setEnabled (info.isFile () && info.suffix () == "m");
-      QAction *load_action = menu.addAction(tr("Load Data"),
-                     this, SLOT(contextmenu_load(bool)));
-      load_action->setEnabled (info.isFile ());
+      if (info.isFile () && info.suffix () == "m")
+        menu.addAction (QIcon (":/actions/icons/artsbuilderexecute.png"),
+                        tr("Run"), this, SLOT(contextmenu_run(bool)));
+      if (info.isFile ())
+        menu.addAction (tr("Load Data"), this, SLOT(contextmenu_load(bool)));
 
       if (info.isDir ())
         {
           menu.addSeparator ();
-          menu.addAction (tr ("Set Current Directory"),
+          menu.addAction (QIcon (":/actions/icons/ok.png"),
+                          tr ("Set Current Directory"),
                           this, SLOT (contextmenu_setcurrentdir (bool)));
         }
 
       menu.addSeparator();
-      menu.addAction(tr("Rename"), this, SLOT(contextmenu_rename(bool)));
-      menu.addAction(QIcon(":/actions/icons/editdelete.png"), tr("Delete"),
-                     this, SLOT(contextmenu_delete(bool)));
+      menu.addAction (tr ("Rename"), this, SLOT(contextmenu_rename(bool)));
+      menu.addAction (QIcon(":/actions/icons/editdelete.png"), tr("Delete"),
+                      this, SLOT(contextmenu_delete(bool)));
 
-      menu.addSeparator();
-      QAction *new_file_action = menu.addAction(
-                   QIcon(":/actions/icons/filenew.png"),
-                   tr("New File"), this, SLOT(contextmenu_newfile(bool)));
-      new_file_action->setEnabled (info.isDir());
-      QAction *new_dir_action  = menu.addAction(
-                   QIcon(":/actions/icons/folder_new.png"),
-                   tr("New Directory"), this, SLOT(contextmenu_newdir(bool)));
-      new_dir_action->setEnabled (info.isDir());
+      if (info.isDir())
+        {
+          menu.addSeparator();
+          menu.addAction (QIcon (":/actions/icons/filenew.png"),
+                          tr ("New File"),
+                          this, SLOT(contextmenu_newfile(bool)));
+          menu.addAction (QIcon (":/actions/icons/folder_new.png"),
+                          tr("New Directory"),
+                          this, SLOT(contextmenu_newdir(bool)));
+        }
 
       menu.exec(_file_tree_view->mapToGlobal(mpos));
 
     }
 }
 
 void
 files_dock_widget::contextmenu_open (bool)
